Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -454,8 +454,13 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
     }
 
     func SELdidClickAddTab() {
-        tabManager?.addTab()
-        presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
+        self.collectionView.performBatchUpdates({ _ in
+            self.tabManager.addTab()
```

Comment:
Its the simplest way I found to know when the insert animation had ended. That's the only reason its there.